src/Microsoft.ComponentDetection.Detectors/yarn/YarnLockComponentDetector.cs

```diff
@@ -69,7 +69,12 @@ private void DetectComponents(YarnLockFile file, string location, ISingleFileCom
             {
                 foreach (var satisfiedVersion in entry.Satisfied)
                 {
-                    yarnPackages.Add($"{entry.Name}@{satisfiedVersion}", entry);
+                    var key = $"{entry.Name}@{satisfiedVersion}";
+                    var addSuccessful = yarnPackages.TryAdd(key, entry);
```

Comment:
changes look good, but perhaps we should bump detector version. I don't think many users are impacted by this, but still out of cautiousness

Reply:
Good idea. Done!